### DISCUSSION TITLE
JSON, XML, YAML, ...

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#1972] [Q&A] [answered]
Is there a Contiki built in lib for parsing/writing one of JSON/XML/YAML/or any other standard?

If there isn't, I'm considering writing a lib and in this case opinions on which of the file formats the community would be most interested in, or what may be best for the OS, are solicited:
- [ ] XML
- [ ] JSON
- [ ] YAML
- [ ] A different one:

**Comment by nvt** [comment, marked as answer]
Yes, there is the os/lib/json library. A CBOR implementation would be good to include in Contiki-NG, but there are some external implementations that can be used.

**Reply by rahav** [reply]
Thank you
